server/rest/dataset.py

```diff
@@ -530,3 +532,35 @@ def createTweetGeocodes(self, item, params):
         .param('id', 'The Item ID', paramType='path')
         .errorResponse('ID was invalid.')
         .errorResponse('Write permission denied on the Item.', 403))
+
+    @access.public
+    @loadmodel(map={'wmsSourceId': 'wmsSource'}, model='item',
+               level=AccessType.READ)
+    def createWmsDataset(self, wmsSource, params):
```

Comment:
Why not create a class wmsDataset that inherits from dataset? And then have all datasets register themselves (in dataset or some other kernel)? In that way we can breakdown the dataset.py into multiple classes.